### PR TITLE
Add plugin for reporting terrain height estimate from the FCU

### PR DESCRIPTION
**Pull request by AndersonRayner**
This plugin is designed to take the `TERRAIN_REPORT` mavlink message and import it into a ros topic.  I've tested it in simulation with ArduCopter 4.1.2 and ROS Noetic and everything seems to work as expected.  I intend on testing in hardware either this week or next week.  

First time contributing so comments welcome :).

### CODE REVIEW

File: mavros_extras/src/plugins/terrain.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+			make_handler(&TerrainPlugin::handle_terrain)
+		};
+	}
+
+private:
+	ros::NodeHandle terrain_nh;
+
+	ros::Publisher terrain_pub;
+
+	void handle_terrain(const mavlink::mavlink_message_t *msg, mavlink::common::msg::TERRAIN_REPORT &terrain) {
```
Comment (vooon) [inline mavros_extras/src/plugins/terrain.cpp:53]: Method name is ok, but i i prefer `handle_{MESSAGE_NAME.lower()}`.

**Review by vooon — COMMENTED** [review]
Looks good, just few things.

File: mavros_extras/src/plugins/terrain.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+public:
+	TerrainPlugin() : PluginBase(),
+		terrain_nh("~terrain")
+	{ }
+
+	void initialize(UAS &uas_) override
+	{
+		PluginBase::initialize(uas_);
+
+		terrain_pub = terrain_nh.advertise<mavros_msgs::Terrain>("in", 10);
```
[the remark this answers is not in the document]
Comment (vooon) [reply mavros_extras/src/plugins/terrain.cpp:38]: Also perhaps better to name message TerrainReport. That plugin might be extended to support terrain data upload.

[the remark this answers is not in the document]
Comment (AndersonRayner) [reply mavros_extras/src/plugins/terrain.cpp:38]: Re naming: Using the `"in"` extension for the advertise follows what the `odom` plugin uses, though happy to change this to whatever (just let me know (or confirm we should go with `~terrain/report`)).  I went with `"in"` to be into ROS (from the FCU), with the vision that (in the future) `"out"` could provide terrain data out of ROS (and to the FCU).

RE: message name: Good suggestion.  Fixed!

File: mavros_extras/src/plugins/terrain.cpp (hunk 1 above)
Comment (AndersonRayner) [reply mavros_extras/src/plugins/terrain.cpp:53]: Good suggestion, fixed!

File: mavros_extras/src/plugins/terrain.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	{
+		return {
+			make_handler(&TerrainPlugin::handle_terrain_report)
+		};
+	}
+
+private:
+	ros::NodeHandle terrain_nh;
+
+	ros::Publisher TerrainReport_pub;
```
Comment (vooon) [inline mavros_extras/src/plugins/terrain.cpp:51]: Please do not use CamelCase for class members. Should be `terrain_report_pub`, or simply `report_pub`.

File: mavros_extras/src/plugins/terrain.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (vooon) [reply mavros_extras/src/plugins/terrain.cpp]: Odom probably not best choice to follow. I think "report" (and rostopic list) would be more intuitive.

File: mavros_extras/src/plugins/terrain.cpp (hunk 3 above)
Comment (AndersonRayner) [reply mavros_extras/src/plugins/terrain.cpp]: Thanks for the feedback.  Fixed.

File: mavros_extras/src/plugins/terrain.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (AndersonRayner) [reply mavros_extras/src/plugins/terrain.cpp]: No worries.  I've switched it over to "report"

**Review by vooon — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**